$OpenBSD: patch-libmariadb_libmariadb_CMakeLists_txt,v 1.7 2020/08/11 13:00:31 sthen Exp $

Remove LIBICONV_PLUG to fix the build with iconv on OpenBSD.

Index: libmariadb/libmariadb/CMakeLists.txt
--- libmariadb/libmariadb/CMakeLists.txt.orig
+++ libmariadb/libmariadb/CMakeLists.txt
@@ -311,7 +311,6 @@ ELSE()
   IF(ICONV_INCLUDE_DIR)
      INCLUDE_DIRECTORIES(BEFORE ${ICONV_INCLUDE_DIR}) 
   ENDIF()
-  ADD_DEFINITIONS(-DLIBICONV_PLUG)
 ENDIF()
 
 IF(ZLIB_FOUND AND WITH_EXTERNAL_ZLIB)
@@ -394,6 +393,8 @@ ENDIF()
 ADD_LIBRARY(mariadbclient STATIC  ${MARIADB_OBJECTS} ${EMPTY_FILE})
 TARGET_LINK_LIBRARIES(mariadbclient ${SYSTEM_LIBS})
 
+SET(LIBVER_OVERRIDE $ENV{LIBmariadb_VERSION})
+
 IF(UNIX)
   ADD_LIBRARY(libmariadb SHARED ${libmariadb_RC} ${MARIADB_OBJECTS} ${EMPTY_FILE})
   SET_TARGET_PROPERTIES(libmariadb PROPERTIES COMPILE_FLAGS "${CMAKE_SHARED_LIBRARY_C_FLAGS}")
@@ -434,17 +435,9 @@ INCLUDE(${CC_SOURCE_DIR}/cmake/symlink.cmake)
 # of the config program. To make sure these programs can
 # use mariadb client library we provide libmysql symlinks
 
-IF(WITH_MYSQLCOMPAT)
-  create_symlink(libmysqlclient${CMAKE_SHARED_LIBRARY_SUFFIX} libmariadb ${INSTALL_LIBDIR})
-  create_symlink(libmysqlclient_r${CMAKE_SHARED_LIBRARY_SUFFIX} libmariadb ${INSTALL_LIBDIR})
-  create_symlink(libmysqlclient${CMAKE_STATIC_LIBRARY_SUFFIX} mariadbclient ${INSTALL_LIBDIR})
-  create_symlink(libmysqlclient_r${CMAKE_STATIC_LIBRARY_SUFFIX} mariadbclient ${INSTALL_LIBDIR})
-ENDIF()
-
-
 SET_TARGET_PROPERTIES(libmariadb PROPERTIES VERSION
- ${CPACK_PACKAGE_VERSION_MAJOR}
- SOVERSION ${CPACK_PACKAGE_VERSION_MAJOR})
+ ${LIBVER_OVERRIDE}
+ SOVERSION ${LIBVER_OVERRIDE})
 
 IF(NOT WIN32)
   SET_TARGET_PROPERTIES(mariadbclient PROPERTIES OUTPUT_NAME "${LIBMARIADB_STATIC_NAME}")
